product_details.py

```diff
@@ -195,23 +197,86 @@ def get_download_url(self, channel, version, platform, locale, force_direct=True
     def platforms(self, channel='release'):
         return self.platform_labels.items()
 
-    def list_releases(self, channel='beta'):
+    def list_releases(self):
         releases = {}
-        for release in self.major_releases:
-            major_version = float(re.findall(r'^\d+\.\d+', release)[0])
+
+        def needs_major_fixup(version_ints: list[int]):
+            """38 started with a point release, so uhhh fix that."""
+            if version_ints[0] != 38:
+                return False
+
+            # If 38.0.1
+            if len(version_ints) >= 2 and version_ints[1] == 0 and version_ints[2] == 1:
+                return True
+
+            return False
+
+        def needs_esr_fixup(version_ints: list[int]):
+            """115.10.2 up until 128.0esr are mislabelled and should be esr builds"""
```

Comment:
Tiny nitpick, feel free to disregard: I see that 128.0esr is handled on line 244 and not in this function, maybe this comment doesn't need to mention it?

Reply:
The intent is to mention that any 115 builds past 115.10.2 need to be labelled as 'esr' instead of 'stability'. 128.0esr is properly labelled. I'm open to better wording here!

Reply:
Ah! No need to reword. 
After stepping away and coming back to it, it's totally clear :+1: